feedback/views.py

```diff
@@ -487,11 +487,18 @@ def update_context_for_feedbacks(request, context, course=None, feedbacks=None,
         )
     )
 
+class PaginatedMixin():
+    def get_paginate_by(self, queryset):
+        value = self.request.GET.get('paginate_by')
+        if value is not None and value in ("50", "100", "200"):
+            return value
+        else:
+            return self.paginate_by
```

Comment:
If the user manually enters an invalid value in the URL GET query parameters, such as `paginate_by=10` or `=foo`, then the invalid value stays in the URL (browser address bar) even when the user browses to the next page in the pagination bar.

I wonder if the `else` branch here in `get_paginate_by()` could reset the GET query parameter to the default value if the current value is invalid. It could be enough to just remove the whole parameter.

Django's `QueryDict` is immutable, but it can be converted to a mutable container.
https://docs.djangoproject.com/en/3.2/ref/request-response/#django.http.QueryDict.__setitem__
```
# Could this work?
q = self.request.GET.copy()
q['paginate_by'] = self.paginate_by
self.request.GET = q

# or
q = self.request.GET.copy()
q.pop('paginate_by')
self.request.GET = q
```
If the above code does not change the URL in the browser address bar, then I guess we have to redirect the user to another URL that uses a valid value for the parameter (or just excludes the `paginate_by` parameter).